File: backend/grafioschtrader-common/src/main/java/grafioschtrader/dto/StockexchangeBaseData.java
package grafioschtrader.dto;

import java.util.List;

import grafioschtrader.entities.Stockexchange;
import grafioschtrader.entities.StockexchangeMic;

public class StockexchangeBaseData {
  public List<Stockexchange> stockexchanges;
  public List<StockexchangeHasSecurity> hasSecurity;
  public List<StockexchangeMic> stockexchangeMics;
  public List<ValueKeyHtmlSelectOptions> countries;
  
  public StockexchangeBaseData(List<Stockexchange> stockexchanges, List<StockexchangeHasSecurity> hasSecurity,
      List<StockexchangeMic> stockexchangeMics, List<ValueKeyHtmlSelectOptions> countries) {
    super();
    this.stockexchanges = stockexchanges;
    this.hasSecurity = hasSecurity;
    this.stockexchangeMics = stockexchangeMics;
    this.countries = countries;
  }
  
}

File: frontend_old/src/app/stockexchange/model/stockexchange.base.data.ts
import {Stockexchange} from '../../entities/stockexchange';
import {StockexchangeHasSecurity} from './stockexchange.has.security';
import {ValueKeyHtmlSelectOptions} from '../../dynamic-form/models/value.key.html.select.options';

export interface StockexchangeBaseData {
  stockexchanges: Stockexchange[];
  hasSecurity: StockexchangeHasSecurity[];
  stockexchangeMics: StockexchangeMic[];
  countries: ValueKeyHtmlSelectOptions[];
}

export interface StockexchangeMic {
  mic: string;
  name: string;
  countryCode: string;
  city?: string;
  website?: string;
  timeZone?: string;
}

